lib/jsdom/living/nodes/HTMLFormElement-impl.js

```diff
@@ -47,14 +47,53 @@ class HTMLFormElementImpl extends HTMLElementImpl {
     super._descendantRemoved.apply(this, arguments);
   }
 
+  _getElementNodes() {
+    if (!this.hasAttributeNS(null, "id")) {
```

Comment:
It looks like this optimization might be invalid. The spec says that the root of the collection is always the root element of the document: https://html.spec.whatwg.org/#dom-form-elements .

In particular, I think you want to consider what happens if you do something like

```js
const elements = form.elements;
form.id = "foo";
// elements should now contain things associated via form=""
```

Since it sounds like that case is not covered in the existing tests, would you mind adding a to-upstream test that fails before fixing that, and succeeds after?

Reply:
Just to double-check here, you want me to:
- test that the HTMLCollection root element is the root of the document
- test that `form.elements` fails to include new elements after changing `form.id`

From what I can find, the HTMLCollection root is never exposed. Not sure how to go about testing that,

Also from what I've found so far in tests, even after updating the HTMLCollection element to the root of the document, the query automatically includes / removes elements on changing `form.id`, as long as `this` in the query is scoped to the HTMLFormElement.